Bottom: 1415561d1bcc6b4377d93082188c72b2fe2d3d11
Top:    8cb60c5ffac8f94ef489a8e73f544789d2e7086f
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2011-12-16 00:56:07 +0000

crosshair.c: Snap to later-tested objects equi-distant with the current snap.

This inequality adjustment means that when an element mark is on-grid, and
the first-snap is a grid location snap, the test against the element mark
will now pass, and mark the snap as a non-grid snap.

This fixes the rather irritating bug where certain footprints, such as SMT
resistors became impossible to grab by their element mark when that mark
landed exactly on a grid point.


---

diff --git a/src/crosshair.c b/src/crosshair.c
index b7bebec..3c6c46e 100644
--- a/src/crosshair.c
+++ b/src/crosshair.c
@@ -819,7 +819,7 @@ check_snap_object (struct snap_data *snap_data, Coord x, Coord y,
   double sq_dist;
 
   sq_dist = crosshair_sq_dist (snap_data->crosshair, x, y);
-  if (sq_dist < snap_data->nearest_sq_dist ||
+  if (sq_dist <= snap_data->nearest_sq_dist ||
       (prefer_to_grid && snap_data->nearest_is_grid && !gui->shift_is_pressed()))
     {
       snap_data->x = x;
